src/app/common/grade-icon/grade-icon.component.ts

```diff
@@ -0,0 +1,25 @@
+import { Component, OnInit, Input, Inject } from '@angular/core';
+import { gradeService } from 'src/app/ajs-upgraded-providers';
+
+@Component({
+  selector: 'grade-icon',
+  templateUrl: 'grade-icon.component.html',
+  styleUrls: ['grade-icon.component.scss'],
+})
+export class GradeIconComponent implements OnInit {
+  @Input() grade: string = 'F';
```

Comment:
Could we make this a union type rather than a string? To limit passing in an invalid grade?

type grade = "F" | "P" | C" ...